lib/src/clojure_lsp/feature/completion.clj

```diff
@@ -327,10 +332,11 @@
     (concat
       (map (fn [item]
              (if-let [snippet-item (get snippet-items-by-label (:label item))]
-               (shared/assoc-some snippet-item
-                                  :detail (:detail item)
-                                  :data (:data item)
-                                  :kind (:kind item))
+               (let [data (shared/assoc-some (:data item)
+                                             :snippet-kind (get completion-kind-enum (:kind item)))]
```

Comment:
I'm kind of lost why we are adding this only because of vim editor. 
This field is no part of the spec and it's already the :kind but as int, right?

Reply:
See my [comment](https://github.com/clojure-lsp/clojure-lsp/pull/720#discussion_r794592439) for why I added it.
As for type: `:kind` will be coerced into an int by the spec + java-lsp lib. I needed to do that manually for `:snippet-kind` because `:data` does not go through the same process.

Regarding the `::`, are you sure? I don't see a spec for `::data` and couldn't find any special handling for `:ns`, `:filename`, or `:name`.

Reply:
I found it. I missed that the key in `:data` just gets to-stringed. Fixing it right now for :snipet-kind.